File: backend/internal/models/project.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// ProjectMember represents a team member in a project
type ProjectMember struct {
	ID       string    `json:"id" bson:"id"`
	Name     string    `json:"name" bson:"name"`
	Email    string    `json:"email" bson:"email"`
	Avatar   string    `json:"avatar,omitempty" bson:"avatar,omitempty"`
	Role     string    `json:"role" bson:"role"` // owner, admin, member, viewer
	Status   string    `json:"status" bson:"status"` // online, away, offline
	LastSeen time.Time `json:"last_seen" bson:"last_seen"`
}

// ProjectTask represents a task within a project
type ProjectTask struct {
	ID          string     `json:"id" bson:"id"`
	Title       string     `json:"title" bson:"title" binding:"required"`
	Description string     `json:"description,omitempty" bson:"description,omitempty"`
	Status      string     `json:"status" bson:"status"` // todo, in_progress, review, completed
	Priority    string     `json:"priority" bson:"priority"` // low, medium, high, urgent
	AssignedTo  string     `json:"assigned_to,omitempty" bson:"assigned_to,omitempty"`
	CreatedAt   time.Time  `json:"created_at" bson:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at" bson:"updated_at"`
	DueDate     *time.Time `json:"due_date,omitempty" bson:"due_date,omitempty"`
	Tags        []string   `json:"tags" bson:"tags"`
	ProjectID   string     `json:"project_id" bson:"project_id"`
}

// ProjectMilestone represents a milestone within a project
type ProjectMilestone struct {
	ID          string    `json:"id" bson:"id"`
	Title       string    `json:"title" bson:"title" binding:"required"`
	Description string    `json:"description,omitempty" bson:"description,omitempty"`
	DueDate     time.Time `json:"due_date" bson:"due_date" binding:"required"`
	Status      string    `json:"status" bson:"status"` // upcoming, in_progress, completed, overdue
	Progress    int       `json:"progress" bson:"progress"` // 0-100
	ProjectID   string    `json:"project_id" bson:"project_id"`
	Tasks       []string  `json:"tasks" bson:"tasks"` // task IDs associated with this milestone
}

// ProjectSettings represents project configuration
type ProjectSettings struct {
	IsPublic         bool `json:"is_public" bson:"is_public"`
	AllowGuestAccess bool `json:"allow_guest_access" bson:"allow_guest_access"`
	AutoSync         bool `json:"auto_sync" bson:"auto_sync"`
	Notifications    bool `json:"notifications" bson:"notifications"`
}

// ProjectStats represents project statistics
type ProjectStats struct {
	TotalTasks     int64     `json:"total_tasks" bson:"total_tasks"`
	CompletedTasks int64     `json:"completed_tasks" bson:"completed_tasks"`
	OverdueTasks   int64     `json:"overdue_tasks" bson:"overdue_tasks"`
	ActiveMembers  int64     `json:"active_members" bson:"active_members"`
	TotalAPICalls  int64     `json:"total_api_calls" bson:"total_api_calls"`
	LastActivity   time.Time `json:"last_activity" bson:"last_activity"`
}

// Project represents a development project
type Project struct {
	ID          primitive.ObjectID  `json:"id" bson:"_id,omitempty"`
	Name        string              `json:"name" bson:"name" binding:"required"`
	Description string              `json:"description,omitempty" bson:"description,omitempty"`
	Type        string              `json:"type" bson:"type" binding:"required"` // api, web, mobile, desktop, data, other
	Status      string              `json:"status" bson:"status"` // planning, active, on_hold, completed, archived
	Priority    string              `json:"priority" bson:"priority"` // low, medium, high, urgent
	Progress    int                 `json:"progress" bson:"progress"` // 0-100
	StartDate   time.Time           `json:"start_date" bson:"start_date"`
	DueDate     *time.Time          `json:"due_date,omitempty" bson:"due_date,omitempty"`
	CreatedAt   time.Time           `json:"created_at" bson:"created_at"`
	UpdatedAt   time.Time           `json:"updated_at" bson:"updated_at"`
	OwnerID     string              `json:"owner_id" bson:"owner_id"`
	Members     []ProjectMember     `json:"members" bson:"members"`
	Tags        []string            `json:"tags" bson:"tags"`
	Milestones  []ProjectMilestone  `json:"milestones" bson:"milestones"`
	Tasks       []ProjectTask       `json:"tasks" bson:"tasks"`
	Settings    ProjectSettings     `json:"settings" bson:"settings"`
	Stats       ProjectStats        `json:"stats" bson:"stats"`
}

// ProjectTemplate represents a project template
type ProjectTemplate struct {
	ID                 primitive.ObjectID `json:"id" bson:"_id,omitempty"`
	Name               string             `json:"name" bson:"name" binding:"required"`
	Description        string             `json:"description" bson:"description"`
	Type               string             `json:"type" bson:"type"` // api, web, mobile, desktop, data, other
	DefaultTasks       []ProjectTask      `json:"default_tasks" bson:"default_tasks"`
	DefaultMilestones  []ProjectMilestone `json:"default_milestones" bson:"default_milestones"`
	Tags               []string           `json:"tags" bson:"tags"`
	EstimatedDuration  int                `json:"estimated_duration" bson:"estimated_duration"` // in days
	CreatedAt          time.Time          `json:"created_at" bson:"created_at"`
	UpdatedAt          time.Time          `json:"updated_at" bson:"updated_at"`
	CreatedBy          string             `json:"created_by" bson:"created_by"`
	IsPublic           bool               `json:"is_public" bson:"is_public"`
}

// ProjectActivity represents activity log entries
type ProjectActivity struct {
	ID          primitive.ObjectID `json:"id" bson:"_id,omitempty"`
	ProjectID   string             `json:"project_id" bson:"project_id"`
	UserID      string             `json:"user_id" bson:"user_id"`
	UserName    string             `json:"user_name" bson:"user_name"`
	Action      string             `json:"action" bson:"action"` // created, updated, deleted, completed, etc.
	EntityType  string             `json:"entity_type" bson:"entity_type"` // project, task, milestone, member
	EntityID    string             `json:"entity_id,omitempty" bson:"entity_id,omitempty"`
	EntityName  string             `json:"entity_name,omitempty" bson:"entity_name,omitempty"`
	Description string             `json:"description" bson:"description"`
	Metadata    map[string]interface{} `json:"metadata,omitempty" bson:"metadata,omitempty"`
	Timestamp   time.Time          `json:"timestamp" bson:"timestamp"`
}

// ProjectFile represents files associated with a project
type ProjectFile struct {
	ID            primitive.ObjectID `json:"id" bson:"_id,omitempty"`
	ProjectID     string             `json:"project_id" bson:"project_id"`
	Name          string             `json:"name" bson:"name" binding:"required"`
	OriginalName  string             `json:"original_name" bson:"original_name"`
	Type          string             `json:"type" bson:"type"` // image, document, code, archive, other
	Size          int64              `json:"size" bson:"size"`
	MimeType      string             `json:"mime_type" bson:"mime_type"`
	URL           string             `json:"url" bson:"url"`
	Description   string             `json:"description,omitempty" bson:"description,omitempty"`
	Tags          []string           `json:"tags" bson:"tags"`
	UploadedBy    string             `json:"uploaded_by" bson:"uploaded_by"`
	UploadedAt    time.Time          `json:"uploaded_at" bson:"uploaded_at"`
	IsShared      bool               `json:"is_shared" bson:"is_shared"`
	DownloadCount int64              `json:"download_count" bson:"download_count"`
	Version       string             `json:"version" bson:"version"`
	Folder        string             `json:"folder,omitempty" bson:"folder,omitempty"`
}

// ProjectComment represents comments on projects, tasks, or milestones
type ProjectComment struct {
	ID          primitive.ObjectID `json:"id" bson:"_id,omitempty"`
	ProjectID   string             `json:"project_id" bson:"project_id"`
	EntityType  string             `json:"entity_type" bson:"entity_type"` // project, task, milestone
	EntityID    string             `json:"entity_id" bson:"entity_id"`
	UserID      string             `json:"user_id" bson:"user_id"`
	UserName    string             `json:"user_name" bson:"user_name"`
	Content     string             `json:"content" bson:"content" binding:"required"`
	CreatedAt   time.Time          `json:"created_at" bson:"created_at"`
	UpdatedAt   time.Time          `json:"updated_at" bson:"updated_at"`
	IsEdited    bool               `json:"is_edited" bson:"is_edited"`
	Mentions    []string           `json:"mentions,omitempty" bson:"mentions,omitempty"` // user IDs mentioned
	Attachments []string           `json:"attachments,omitempty" bson:"attachments,omitempty"` // file IDs
}

// ProjectTimeEntry represents time tracking entries
type ProjectTimeEntry struct {
	ID          primitive.ObjectID `json:"id" bson:"_id,omitempty"`
	ProjectID   string             `json:"project_id" bson:"project_id"`
	TaskID      string             `json:"task_id,omitempty" bson:"task_id,omitempty"`
	UserID      string             `json:"user_id" bson:"user_id"`
	UserName    string             `json:"user_name" bson:"user_name"`
	Description string             `json:"description,omitempty" bson:"description,omitempty"`
	StartTime   time.Time          `json:"start_time" bson:"start_time"`
	EndTime     *time.Time         `json:"end_time,omitempty" bson:"end_time,omitempty"`
	Duration    int64              `json:"duration" bson:"duration"` // in seconds
	IsRunning   bool               `json:"is_running" bson:"is_running"`
	CreatedAt   time.Time          `json:"created_at" bson:"created_at"`
	UpdatedAt   time.Time          `json:"updated_at" bson:"updated_at"`
}

// ProjectAnalytics represents analytics data for a project
type ProjectAnalytics struct {
	ProjectID         string                 `json:"project_id" bson:"project_id"`
	DateRange         DateRange              `json:"date_range" bson:"date_range"`
	TaskCompletion    TaskCompletionMetrics  `json:"task_completion" bson:"task_completion"`
	TeamActivity      []TeamActivityMetrics  `json:"team_activity" bson:"team_activity"`
	TimeTracking      []TimeTrackingMetrics  `json:"time_tracking" bson:"time_tracking"`
	MilestoneProgress []MilestoneMetrics     `json:"milestone_progress" bson:"milestone_progress"`
	VelocityMetrics   []VelocityMetrics      `json:"velocity_metrics" bson:"velocity_metrics"`
	Priority          PriorityDistribution   `json:"priority_distribution" bson:"priority_distribution"`
	GeneratedAt       time.Time              `json:"generated_at" bson:"generated_at"`
}

type DateRange struct {
	Start time.Time `json:"start" bson:"start"`
	End   time.Time `json:"end" bson:"end"`
}

type TaskCompletionMetrics struct {
	Completed  int `json:"completed" bson:"completed"`
	InProgress int `json:"in_progress" bson:"in_progress"`
	Todo       int `json:"todo" bson:"todo"`
	Review     int `json:"review" bson:"review"`
}

type TeamActivityMetrics struct {
	UserID         string  `json:"user_id" bson:"user_id"`
	UserName       string  `json:"user_name" bson:"user_name"`
	TasksCompleted int     `json:"tasks_completed" bson:"tasks_completed"`
	HoursLogged    float64 `json:"hours_logged" bson:"hours_logged"`
	LastActive     time.Time `json:"last_active" bson:"last_active"`
}

type TimeTrackingMetrics struct {
	Date  string  `json:"date" bson:"date"`
	Hours float64 `json:"hours" bson:"hours"`
}

type MilestoneMetrics struct {
	ID       string    `json:"id" bson:"id"`
	Name     string    `json:"name" bson:"name"`
	Progress int       `json:"progress" bson:"progress"`
	DueDate  time.Time `json:"due_date" bson:"due_date"`
	Status   string    `json:"status" bson:"status"`
}

type VelocityMetrics struct {
	Week           string  `json:"week" bson:"week"`
	TasksCompleted int     `json:"tasks_completed" bson:"tasks_completed"`
	AverageTime    float64 `json:"average_time" bson:"average_time"` // hours
}

type PriorityDistribution struct {
	Urgent int `json:"urgent" bson:"urgent"`
	High   int `json:"high" bson:"high"`
	Medium int `json:"medium" bson:"medium"`
	Low    int `json:"low" bson:"low"`
}

File: frontend/src/lib/stores/project.ts
import { writable } from 'svelte/store';

export interface ProjectTask {
    id: string;
    title: string;
    description?: string;
    status: 'todo' | 'in_progress' | 'review' | 'completed';
    priority: 'low' | 'medium' | 'high' | 'urgent';
    assignedTo?: string;
    createdAt: Date;
    updatedAt: Date;
    dueDate?: Date;
    tags: string[];
    projectId: string;
}

export interface ProjectMilestone {
    id: string;
    title: string;
    description?: string;
    dueDate: Date;
    status: 'upcoming' | 'in_progress' | 'completed' | 'overdue';
    progress: number; // 0-100
    projectId: string;
    tasks: string[]; // task IDs
}

export interface ProjectMember {
    id: string;
    name: string;
    email: string;
    avatar?: string;
    role: 'owner' | 'admin' | 'member' | 'viewer';
    status: 'online' | 'away' | 'offline';
    lastSeen: Date;
}

export interface Project {
    id: string;
    name: string;
    description?: string;
    type: 'api' | 'web' | 'mobile' | 'desktop' | 'data' | 'other';
    status: 'planning' | 'active' | 'on_hold' | 'completed' | 'archived';
    priority: 'low' | 'medium' | 'high' | 'urgent';
    progress: number; // 0-100
    startDate: Date;
    dueDate?: Date;
    createdAt: Date;
    updatedAt: Date;
    ownerId: string;
    members: ProjectMember[];
    tags: string[];
    milestones: ProjectMilestone[];
    tasks: ProjectTask[];
    settings: {
        isPublic: boolean;
        allowGuestAccess: boolean;
        autoSync: boolean;
        notifications: boolean;
    };
    stats: {
        totalTasks: number;
        completedTasks: number;
        overdueTasks: number;
        activeMembers: number;
        totalApiCalls: number;
        lastActivity: Date;
    };
}

export interface ProjectTemplate {
    id: string;
    name: string;
    description: string;
    type: Project['type'];
    defaultTasks: Omit<ProjectTask, 'id' | 'projectId' | 'createdAt' | 'updatedAt'>[];
    defaultMilestones: Omit<ProjectMilestone, 'id' | 'projectId' | 'tasks'>[];
    tags: string[];
    estimatedDuration: number; // in days
}

interface ProjectStoreState {
    projects: Project[];
    currentProject: Project | null;
    tasks: ProjectTask[];
    milestones: ProjectMilestone[];
    templates: ProjectTemplate[];
    isLoading: boolean;
    error: string | null;
    filters: {
        status: Project['status'] | 'all';
        type: Project['type'] | 'all';
        priority: Project['priority'] | 'all';
        search: string;
    };
    view: 'grid' | 'list' | 'kanban';
}

const initialState: ProjectStoreState = {
    projects: [],
    currentProject: null,
    tasks: [],
    milestones: [],
    templates: [],
    isLoading: false,
    error: null,
    filters: {
        status: 'all',
        type: 'all',
        priority: 'all',
        search: ''
    },
    view: 'grid'
};

function createProjectStore() {
    const { subscribe, set, update } = writable(initialState);

    return {
        subscribe,
        
        // Project actions
        setProjects: (projects: Project[]) => 
            update(state => ({ ...state, projects })),
            
        addProject: (project: Project) =>
            update(state => ({ 
                ...state, 
                projects: [...state.projects, project] 
            })),
            
        updateProject: (projectId: string, updates: Partial<Project>) =>
            update(state => ({
                ...state,
                projects: state.projects.map(p => 
                    p.id === projectId ? { ...p, ...updates, updatedAt: new Date() } : p
                ),
                currentProject: state.currentProject?.id === projectId 
                    ? { ...state.currentProject, ...updates, updatedAt: new Date() }
                    : state.currentProject
            })),
            
        deleteProject: (projectId: string) =>
            update(state => ({
                ...state,
                projects: state.projects.filter(p => p.id !== projectId),
                currentProject: state.currentProject?.id === projectId ? null : state.currentProject
            })),
            
        setCurrentProject: (project: Project | null) =>
            update(state => ({ ...state, currentProject: project })),
            
        // Task actions
        setTasks: (tasks: ProjectTask[]) =>
            update(state => ({ ...state, tasks })),
            
        addTask: (task: ProjectTask) =>
            update(state => ({ 
                ...state, 
                tasks: [...state.tasks, task],
                projects: state.projects.map(p => 
                    p.id === task.projectId 
                        ? { 
                            ...p, 
                            tasks: [...p.tasks, task],
                            stats: { 
                                ...p.stats, 
                                totalTasks: p.stats.totalTasks + 1,
                                lastActivity: new Date()
                            }
                        }
                        : p
                )
            })),
            
        updateTask: (taskId: string, updates: Partial<ProjectTask>) =>
            update(state => {
                const updatedTask = state.tasks.find(t => t.id === taskId);
                if (!updatedTask) return state;
                
                const newTask = { ...updatedTask, ...updates, updatedAt: new Date() };
                const statusChanged = updatedTask.status !== newTask.status;
                
                return {
                    ...state,
                    tasks: state.tasks.map(t => t.id === taskId ? newTask : t),
                    projects: state.projects.map(p => {
                        if (p.id === newTask.projectId) {
                            const updatedTasks = p.tasks.map(t => t.id === taskId ? newTask : t);
                            const completedTasks = updatedTasks.filter(t => t.status === 'completed').length;
                            const overdueTasks = updatedTasks.filter(t => 
                                t.dueDate && new Date(t.dueDate) < new Date() && t.status !== 'completed'
                            ).length;
                            
                            return {
                                ...p,
                                tasks: updatedTasks,
                                stats: {
                                    ...p.stats,
                                    completedTasks: statusChanged ? completedTasks : p.stats.completedTasks,
                                    overdueTasks: statusChanged ? overdueTasks : p.stats.overdueTasks,
                                    lastActivity: new Date()
                                }
                            };
                        }
                        return p;
                    })
                };
            }),
            
        deleteTask: (taskId: string) =>
            update(state => {
                const task = state.tasks.find(t => t.id === taskId);
                if (!task) return state;
                
                return {
                    ...state,
                    tasks: state.tasks.filter(t => t.id !== taskId),
                    projects: state.projects.map(p => 
                        p.id === task.projectId
                            ? {
                                ...p,
                                tasks: p.tasks.filter(t => t.id !== taskId),
                                stats: {
                                    ...p.stats,
                                    totalTasks: p.stats.totalTasks - 1,
                                    completedTasks: task.status === 'completed' 
                                        ? p.stats.completedTasks - 1 
                                        : p.stats.completedTasks,
                                    lastActivity: new Date()
                                }
                            }
                            : p
                    )
                };
            }),
            
        // Milestone actions
        setMilestones: (milestones: ProjectMilestone[]) =>
            update(state => ({ ...state, milestones })),
            
        addMilestone: (milestone: ProjectMilestone) =>
            update(state => ({ 
                ...state, 
                milestones: [...state.milestones, milestone],
                projects: state.projects.map(p => 
                    p.id === milestone.projectId
                        ? { ...p, milestones: [...p.milestones, milestone] }
                        : p
                )
            })),
            
        updateMilestone: (milestoneId: string, updates: Partial<ProjectMilestone>) =>
            update(state => ({
                ...state,
                milestones: state.milestones.map(m => 
                    m.id === milestoneId ? { ...m, ...updates } : m
                ),
                projects: state.projects.map(p => ({
                    ...p,
                    milestones: p.milestones.map(m => 
                        m.id === milestoneId ? { ...m, ...updates } : m
                    )
                }))
            })),
            
        // Filter actions
        setFilters: (filters: Partial<ProjectStoreState['filters']>) =>
            update(state => ({ 
                ...state, 
                filters: { ...state.filters, ...filters } 
            })),
            
        setView: (view: ProjectStoreState['view']) =>
            update(state => ({ ...state, view })),
            
        // Loading and error states
        setLoading: (isLoading: boolean) =>
            update(state => ({ ...state, isLoading })),
            
        setError: (error: string | null) =>
            update(state => ({ ...state, error })),
            
        // Templates
        setTemplates: (templates: ProjectTemplate[]) =>
            update(state => ({ ...state, templates })),
            
        // Utility functions
        getProjectById: (projectId: string): Project | undefined => {
            let project: Project | undefined;
            update(state => {
                project = state.projects.find(p => p.id === projectId);
                return state;
            });
            return project;
        },
        
        getTasksByProject: (projectId: string): ProjectTask[] => {
            let tasks: ProjectTask[] = [];
            update(state => {
                tasks = state.tasks.filter(t => t.projectId === projectId);
                return state;
            });
            return tasks;
        },
        
        getMilestonesByProject: (projectId: string): ProjectMilestone[] => {
            let milestones: ProjectMilestone[] = [];
            update(state => {
                milestones = state.milestones.filter(m => m.projectId === projectId);
                return state;
            });
            return milestones;
        }
    };
}

export const projectStore = createProjectStore();

// Default project templates
export const defaultTemplates: ProjectTemplate[] = [
    {
        id: 'api-project',
        name: 'API Development',
        description: 'RESTful API development with testing and documentation',
        type: 'api',
        defaultTasks: [
            {
                title: 'Design API Architecture',
                description: 'Define endpoints, data models, and authentication',
                status: 'todo',
                priority: 'high',
                tags: ['planning', 'architecture']
            },
            {
                title: 'Setup Development Environment',
                description: 'Configure database, server, and testing tools',
                status: 'todo',
                priority: 'high',
                tags: ['setup', 'devops']
            },
            {
                title: 'Implement Core Endpoints',
                description: 'Build CRUD operations for main resources',
                status: 'todo',
                priority: 'medium',
                tags: ['development', 'backend']
            },
            {
                title: 'Add Authentication & Authorization',
                description: 'Implement JWT-based authentication system',
                status: 'todo',
                priority: 'high',
                tags: ['security', 'auth']
            },
            {
                title: 'Write API Documentation',
                description: 'Create comprehensive API documentation',
                status: 'todo',
                priority: 'medium',
                tags: ['documentation']
            },
            {
                title: 'Setup API Testing',
                description: 'Create test collections and automated tests',
                status: 'todo',
                priority: 'medium',
                tags: ['testing', 'qa']
            }
        ],
        defaultMilestones: [
            {
                title: 'MVP API Complete',
                description: 'Core functionality implemented and tested',
                dueDate: new Date(Date.now() + 30 * 24 * 60 * 60 * 1000), // 30 days
                status: 'upcoming',
                progress: 0
            },
            {
                title: 'Production Ready',
                description: 'Security, documentation, and deployment complete',
                dueDate: new Date(Date.now() + 60 * 24 * 60 * 60 * 1000), // 60 days
                status: 'upcoming',
                progress: 0
            }
        ],
        tags: ['api', 'backend', 'rest'],
        estimatedDuration: 45
    },
    {
        id: 'web-app',
        name: 'Web Application',
        description: 'Full-stack web application development',
        type: 'web',
        defaultTasks: [
            {
                title: 'Project Setup & Architecture',
                description: 'Setup frontend, backend, and database',
                status: 'todo',
                priority: 'high',
                tags: ['setup', 'architecture']
            },
            {
                title: 'Design System & UI Components',
                description: 'Create reusable UI components and design system',
                status: 'todo',
                priority: 'medium',
                tags: ['design', 'frontend']
            },
            {
                title: 'User Authentication',
                description: 'Implement user registration, login, and session management',
                status: 'todo',
                priority: 'high',
                tags: ['auth', 'security']
            },
            {
                title: 'Core Features Development',
                description: 'Build main application features',
                status: 'todo',
                priority: 'high',
                tags: ['development', 'features']
            },
            {
                title: 'Testing & Quality Assurance',
                description: 'Unit tests, integration tests, and manual testing',
                status: 'todo',
                priority: 'medium',
                tags: ['testing', 'qa']
            },
            {
                title: 'Deployment & DevOps',
                description: 'Setup CI/CD pipeline and production deployment',
                status: 'todo',
                priority: 'medium',
                tags: ['devops', 'deployment']
            }
        ],
        defaultMilestones: [
            {
                title: 'Alpha Version',
                description: 'Core features implemented for internal testing',
                dueDate: new Date(Date.now() + 45 * 24 * 60 * 60 * 1000), // 45 days
                status: 'upcoming',
                progress: 0
            },
            {
                title: 'Beta Release',
                description: 'Feature-complete version for user testing',
                dueDate: new Date(Date.now() + 75 * 24 * 60 * 60 * 1000), // 75 days
                status: 'upcoming',
                progress: 0
            },
            {
                title: 'Production Launch',
                description: 'Stable version ready for public release',
                dueDate: new Date(Date.now() + 90 * 24 * 60 * 60 * 1000), // 90 days
                status: 'upcoming',
                progress: 0
            }
        ],
        tags: ['web', 'fullstack', 'frontend', 'backend'],
        estimatedDuration: 90
    },
    {
        id: 'mobile-app',
        name: 'Mobile Application',
        description: 'Cross-platform mobile application development',
        type: 'mobile',
        defaultTasks: [
            {
                title: 'Mobile App Architecture',
                description: 'Define app structure, navigation, and state management',
                status: 'todo',
                priority: 'high',
                tags: ['architecture', 'planning']
            },
            {
                title: 'UI/UX Design Implementation',
                description: 'Create mobile-responsive design and components',
                status: 'todo',
                priority: 'high',
                tags: ['design', 'ui', 'mobile']
            },
            {
                title: 'API Integration',
                description: 'Connect mobile app to backend services',
                status: 'todo',
                priority: 'medium',
                tags: ['api', 'integration']
            },
            {
                title: 'Offline Functionality',
                description: 'Implement local storage and offline capabilities',
                status: 'todo',
                priority: 'medium',
                tags: ['offline', 'storage']
            },
            {
                title: 'Testing on Devices',
                description: 'Test on various mobile devices and screen sizes',
                status: 'todo',
                priority: 'medium',
                tags: ['testing', 'devices']
            },
            {
                title: 'App Store Deployment',
                description: 'Prepare and submit to app stores',
                status: 'todo',
                priority: 'low',
                tags: ['deployment', 'stores']
            }
        ],
        defaultMilestones: [
            {
                title: 'Prototype Ready',
                description: 'Basic functionality working on mobile devices',
                dueDate: new Date(Date.now() + 30 * 24 * 60 * 60 * 1000), // 30 days
                status: 'upcoming',
                progress: 0
            },
            {
                title: 'Beta Testing',
                description: 'Feature-complete app ready for beta testing',
                dueDate: new Date(Date.now() + 60 * 24 * 60 * 60 * 1000), // 60 days
                status: 'upcoming',
                progress: 0
            },
            {
                title: 'App Store Release',
                description: 'App published and available for download',
                dueDate: new Date(Date.now() + 90 * 24 * 60 * 60 * 1000), // 90 days
                status: 'upcoming',
                progress: 0
            }
        ],
        tags: ['mobile', 'cross-platform', 'app'],
        estimatedDuration: 75
    }
];